dts/riscv/it8xxx2-wuc-map.dtsi

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) 2022 ITE Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/dt-util.h>
+
+/ {
+	/* Mapping between wuc bits and source device */
+	it8xxx2-wuc-map {
+		compatible = "ite,it8xxx2-wuc-map";
```

Comment:
Will this `wuc-map` replace the `gpio_irqs[]` [table](https://github.com/zephyrproject-rtos/zephyr/blob/f4a0ea7b43eee4d2ee735ab6beccc68c9d40a7d0/drivers/gpio/gpio_ite_it8xxx2.c#L119)?

Reply:
I think we still need gpio_irqs[].gpio_pin member to find out which GPIO_INT is fired in gpio_ite_isr(). And we needn't the reset of gpio_irqs[] members. 
There is a draft CL about GPIO changes for WUC:
https://chromium-review.googlesource.com/c/chromiumos/third_party/zephyr/+/3359139/3

Reply:
You could initialize the gpio_irqs[] table using the `it8xxx2-wuc-map`.  All the WUC related mappings would be contained in one place.
```
	it8xxx2-wuc-map {
		wuc_wu22: wu22 {
			wucs = <&wuc2 2 BIT(2)>; /* WUI2 */
			gpios = <&gpioc 4 0>;  /* GPC4 */
		};
```

Alternately you could add another cell to `wuc-cells`:
```
		wuc_wu22: wu22 {
			wucs = <&wuc2 2 BIT(2) 4>; /* WUI2, GPC4 */
		};
```

Reply:
Ack.
I find that the gpio pin can be found by comparing the irq and interrupt property, so we needn't the gpio_irqs[] table.
https://chromium-review.googlesource.com/c/chromiumos/third_party/zephyr/+/3359139/4